python/pyspark/pandas/data_type_ops/base.py

```diff
@@ -314,9 +320,11 @@ def __or__(self, left: IndexOpsLike, right: Any) -> SeriesOrIndex:
         raise TypeError("Bitwise or can not be applied to %s." % self.pretty_name)
 
     def rand(self, left: IndexOpsLike, right: Any) -> SeriesOrIndex:
```

Comment:
As @HyukjinKwon mentioned in https://github.com/apache/spark/pull/33581/files#r680433058,

I think maybe we can just fix the typing from `right: Any` to something like `right: Union[List, Tuple, Dict, Set]` rather adding function ?? (Or we can take both typing and function ??)

I have no string opinion, too, but anyway the typing here `right: Any` seems wrong

Reply:
We disallow `List, Tuple, Dict, Set` for `right`. The function added is to raise a TypeError for these list-likes. Sorry if there is any confusion.

Reply:
Oh, yeah I got it. Looks good to me as is.